src/server/SocketServer.java

```diff
@@ -106,34 +106,30 @@ public void setup() {
 
     /**
      * Starts the main server loop to accept incoming client connections.
-     * This method runs continuously, accepting new client connections and
-     * spawning ServerHandler threads for each connected client.
+     * This method runs continuously, accepting one new client connection
+     * in each loop iteration and spawning a new ServerHandler thread for it.
      */
     public void startAcceptingRequest() {
         if (this.serverSocket == null) {
             LOGGER.error("Cannot start accepting requests. Server has not been initialized.");
             return;
         }
 
-        try {
-            // Player 1 connection
-            LOGGER.info("player 1 is connecting....");
-            Socket player1Socket = serverSocket.accept();
-
-            ServerHandler player1Handler = new ServerHandler(player1Socket, "Player1");
-            player1Handler.start();
+        LOGGER.info("Server is now accepting connections from multiple clients...");
 
-            // Player 2 connection
-            LOGGER.info("player 2 is connecting....");
-            Socket player2Socket = serverSocket.accept();
-
-            ServerHandler player2Handler = new ServerHandler(player2Socket, "Player2");
-            player2Handler.start();
+        try {
+            while (true) { // Infinite loop — server runs continuously
+                Socket clientSocket = serverSocket.accept();
+                LOGGER.info("New client connected from {}", clientSocket.getInetAddress());
```

Comment:
Lets add the port number too since there are likely to be multiple connections from 127.0.0.1 when using Android emulators
```suggestion
                LOGGER.info("New client connected from {}:{}", clientSocket.getInetAddress(), clientSocket.getPort());

```